core/services/ocr2/delegate.go

```diff
@@ -588,8 +589,26 @@ func (d *Delegate) newServicesGenericPlugin(
 		OffchainConfigDigester:       provider.OffchainConfigDigester(),
 	}
 
+	var envVars []string
+	switch p.PluginName {
+	case "median":
```

Comment:
@jmank88 `newServicesGenericPlugin` shouldn't have any plugin-specific code in it. Can we introduce a new variable, eg. GenericPluginEnv or similar, and set that instead?

Reply:
Well this is analogous to the implicit command resolution that is already being done with `chailink-<PluginName>`. How would you propose that we handle it?

Reply:
How about deriving the env var name from the pluginName, eg.

```
Var(fmt.Sprintf("CL_%s_ENV", strings.Upper(p.PluginName)))
```

We should probably also allow this to be overridden in the config by adding a relevant option in pluginConfig.

Reply:
I think we'll be making a mess if we just derive everything dynamically like this. For example, we are trying to use `feeds` instead of `median`, and that may have already broken `func defaultPathFromPluginName`.

Reply:
I think we would at least have to align things to do one look-up, and then associate everything with that. Rather than doing disconnected logic in different places. i.e. `median` -> `chainlink-feeds` and then we find the env associated with `chainlink-feeds`.

Reply:
No, just that they need to be aligned in one place instead of making assumptions and transforming values in separate places. Need to take two steps back though since things already seem inconsistent.

Reply:
I think I came up with something coherent. `defaultPathFromPluginName` bypassing the env vars was the part that didn't fit. Now both this and that logic are aligned through an `env.Plugin`.